src/a98/types.d.ts

```diff
@@ -0,0 +1,12 @@
+interface A98Mode {
+	mode: 'a98';
+}
+
+export interface A98 {
+	r: number;
+	g: number;
+	b: number;
+	alpha?: number;
+}
+
+export interface A98WithMode extends A98, A98Mode {}
```

Comment:
It doesn't seem like the `*Mode` interfaces are used for anything but this extension (per each colorspace), so I wonder if it would be neater to have an utility type à la:

```typescript
type WithMode<C, Mode> = C & {mode: Mode};
```
with use like
```typescript
export interface A98WithMode = WithMode<A98, 'a98'>;
```

That could technically be extended to something like

```typescript
// utilities
type ColorType<Components extends string> = {[component in Components]: number} & {alpha?: number};
type WithMode<C, Mode extends string> = C & {mode: Mode};

type A98 = ColorType<'r' | 'g' | 'b'>;
type A98WithMode = WithMode<A98, 'a98'>;
```
but I'm not sure if the types we'd end up with then are very readable.

Secondly, there's a finite number of channel configurations in the project (RGB, LCH, LAB, YIQ, a couple more); would it be DRY-er to declare those in one place and reuse them for the colorspaces that use them?

Reply:
![Screenshot from 2022-04-12 23-00-06](https://user-images.githubusercontent.com/102924398/163053809-a1547dd9-ee32-436a-ac98-efe507ab958f.png)

I think we should follow project conventions, this is why I put each type declaration near to `definition.js` files in each color space folder.

The second thing to consider is that 'mode' property is not necessary in convert functions.
![Screenshot from 2022-04-12 23-04-34](https://user-images.githubusercontent.com/102924398/163054466-fb403529-a80d-4be3-b491-320552cfe0c0.png)
 This is the reason why I choose to add color types `WithMode` and color types without:
![Screenshot from 2022-04-12 23-06-09](https://user-images.githubusercontent.com/102924398/163054701-00eabe94-9d94-4be5-af99-10736fcb6d56.png)
I think I can use `Omit<ConcreteColorWithMode, 'mode'>` in all conversion functions to simplify things. I will try this.

TODO:
- [ ] try simplifying types to get rid of ugly endings in names

Reply:
@akx here: https://github.com/Evercoder/culori/pull/166/commits/a147302f841f68f0ca30640cf8ac1e0f78c2e8fa